lib/rock/webapp/tasks/api.rb

```diff
@@ -156,24 +156,31 @@ def get_operation(name_service, name, operation_name)
                         optional :binary, type: Boolean, default: false
                         optional :init, type: Boolean, default: false
                     end
-                    get ':name_service/:name/ports/:port_name/read', requirements: { name_service: ValidHostnameRegex } do
-
-                        port = get_port(*params.values_at('name_service', 'name', 'port_name', 'init', 'timeout'))
-                        #port = port_by_task_and_name(*params.values_at('name_service', 'name', 'port_name'))
-
-                        if !port.is_reader?
-                            error! "#{port.name} is an input port, cannot read"
-                        end
 
+                    get ':name_service/:name/ports/:port_name/read', requirements: { name_service: ValidHostnameRegex } do
+                    params do
+                        optional :proxy, type: Boolean, default: false
+                    end
                         if Faye::WebSocket.websocket?(env)
-                            port = port.port.to_async.reader(init: true, pull: true)
+                            if params["proxy"]
+                                proxy = Orocos::Async.proxy("#{params[:name_service]}/#{params[:name]}")
```

Comment:
Tested it manually, it actually works.

I guess the Async.proxy constructor supports the *